cmd/entrypoint/runner_test.go

```diff
@@ -14,9 +15,22 @@ func TestRealRunnerSignalForwarding(t *testing.T) {
 	rr := realRunner{}
 	rr.signals = make(chan os.Signal, 1)
 	rr.signals <- syscall.SIGINT
-	if err := rr.Run("sleep", "3600"); err.Error() == "signal: interrupt" {
+	if err := rr.Run("", "sleep", "3600"); err.Error() == "signal: interrupt" {
 		t.Logf("SIGINT forwarded to Entrypoint")
 	} else {
 		t.Fatalf("Unexpected error received: %v", err)
 	}
 }
+
+// TestRealRunnerTimeout tests whether cmd is killed after a millisecond even though it's supposed to sleep for 10 milliseconds.
+func TestRealRunnerTimeout(t *testing.T) {
+	rr := realRunner{}
+	timeout := "1ms"
+	if err := rr.Run(timeout, "sleep", "0.01"); err != nil {
```

Comment:
It might be worth adding a couple of tests to codify how the function deals with edge cases like: invalid timeout string, negative timeout string, empty string.

Reply:
Agreed on the empty string.
As for the invalid/negative timeout string, I'm not sure it adds much considering that the [webhook](https://github.com/tektoncd/pipeline/blob/d7cfc6671dae95c935331a20189d521b1a8379ac/pkg/apis/pipeline/v1beta1/task_validation.go#L164) would already bar such strings in the first place, what do you think?

Reply:
Hm, I think there are still edge cases that wouldn't be caught by the validating webhook - the controller could have a bug in it that mangles the strings it sends in the entrypoint args, another developer could accidentally introduce a "-" to the duration value.

Edge cases like this can seem a bit silly but it's better (in my view anyway) for the code to fail loudly if we introduce any bugs like that than for it to start quietly exhibiting inexplicable behaviour that we don't have testing around.

Reply:
I  understand. As you pointed out offline, this will be resolved in #3304 .

Reply:
Additionally I added unit tests for the webhook.